### PR TITLE
NaifSpice and sensor_type refactor

### PR DESCRIPTION
**Pull request by Kelvinrr**
Changed to address #132 and #129. Still, haven't found a clean way to do naif codes. Need something that can list all naif keys used in the Spice class.

### CODE REVIEW

File: ale/base/data_naif.py [hunk 1]
```diff
@@ -118,8 +118,9 @@ def _sun_position(self):
 
     @property
     def _sun_velocity(self):
+        center_time = self.ephemeris_time
```
Comment (jessemapel) [inline ale/base/data_naif.py]: Is this leftover?

File: ale/base/data_naif.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def reference_frame(self):
     @property
     def _sun_position(self):
         sun_state, _ = spice.spkezr("SUN",
-                                     self.center_ephemeris_time,
+                                     self.center_time,
```
Comment (jessemapel) [inline ale/base/data_naif.py]: Do we need `center_time` as part of the base Driver interface?

File: ale/base/label_pds3.py [hunk 3]
```diff
@@ -61,26 +61,19 @@ def target_name(self):
...
         return spice.bodn2c(self.label['TARGET_NAME'])
 
     @property
-    def starting_ephemeris_time(self):
-        if not hasattr(self, '_starting_ephemeris_time'):
-            sclock = self.label['SPACECRAFT_CLOCK_START_COUNT']
-            self._starting_ephemeris_time = spice.scs2e(self.spacecraft_id, sclock)
-        return self._starting_ephemeris_time
+    def _starting_ephemeris_time(self):
+        return self.label['SPACECRAFT_CLOCK_START_COUNT']
```
Comment (jessemapel) [inline ale/base/label_pds3.py]: It makes sense to have the clock count be provided by the label and then the SPICE mix-in can convert it to an ephemeris time.

File: ale/base/label_pds3.py [hunk 4]
```diff
@@ -61,26 +61,19 @@ def target_name(self):
...
 
     @property
-    def spacecraft_clock_stop_count(self):
-        sc = self.label.get('SPACECRAFT_CLOCK_STOP_COUNT', None)
-        if sc == 'N/A':
-            sc = None
-        return sc
+    def _ending_ephemeris_time(self):
+        return self.label['SPACECRAFT_CLOCK_STOP_COUNT']
```
Comment (jessemapel) [inline ale/base/label_pds3.py]: Same as `_starting_ephemeris_time` this should stay `spacecraft_clock_stop_count` and then the SPICE mix-in can provide `_ending_ephemeris_time`

File: ale/base/label_pds3.py (hunk 3 above)
Comment (jessemapel) [reply ale/base/label_pds3.py]: With the current naming, the scs2e calls in SPICE don't make sense, it looks like converting ephemeris time to ephemeris time with a method that converts clock strings to ephemeris time.

File: ale/base/data_naif.py (hunk 2 above)
Comment (Kelvinrr) [reply ale/base/data_naif.py]: Doesn't CSM use center time as acquisition time for framers?

File: ale/base/label_pds3.py (hunk 3 above)
Comment (Kelvinrr) [reply ale/base/label_pds3.py]: agree that at least better naming is needed. Not sure what though.

File: ale/base/data_naif.py (hunk 1 above)
Comment (Kelvinrr) [reply ale/base/data_naif.py]: yeah, should be removed. Good catch.

File: ale/base/data_naif.py (hunk 2 above)
Comment (jessemapel) [reply ale/base/data_naif.py]: It does, we have it in the Framer mix-in right now, should it be across all of the sensor_types? It looks like yes.

File: ale/base/data_naif.py [hunk 5]
```diff
@@ -129,70 +129,68 @@ def _sun_velocity(self):
...
-            if fn == 'N/A':
-                fn = 0
-        else:
-            fn = 0
+    def _detector_center_sample(self):
+        return float(spice.gdpool('INS{}_BORESIGHT_SAMPLE'.format(self.ikid), 0, 1)[0])
 
-        return self.ikid - int(fn)
+    @property
+    def _detector_center_line(self):
```
Comment (jessemapel) [inline ale/base/data_naif.py]: Should be public, not private

File: ale/base/data_naif.py [hunk 6]
```diff
@@ -129,70 +129,68 @@ def _sun_velocity(self):
...
 
     @property
-    def fikid(self):
-        if isinstance(self, Framer):
-            fn = self.filter_number
-            if fn == 'N/A':
-                fn = 0
-        else:
-            fn = 0
+    def _detector_center_sample(self):
```
Comment (jessemapel) [inline ale/base/data_naif.py]: Should be public, not private

File: ale/drivers/kaguya_drivers.py [hunk 7]
```diff
@@ -69,21 +69,20 @@ def _tc_id(self):
         return spice.bods2c("LISM_{}".format(self.label.get("INSTRUMENT_ID")))
 
     @property
-    def ending_ephemeris_time(self):
-        if not hasattr(self, '_ending_ephemeris_time'):
-            # We need to get the corrected time
-            self._ending_ephemeris_time = self.label.get('CORRECTED_SC_CLOCK_STOP_COUNT').value
-            self._ending_ephemeris_time = spice.sct2e(self.spacecraft_id, self._ending_ephemeris_time)
-        return self._ending_ephemeris_time
+    def _ending_ephemeris_time(self):
```
Comment (jessemapel) [inline ale/drivers/kaguya_drivers.py]: Should be public, not private; so that it properly overrides.

File: ale/drivers/kaguya_drivers.py [hunk 8]
```diff
@@ -69,21 +69,20 @@ def _tc_id(self):
...
+        return self.label.get('CORRECTED_SC_CLOCK_STOP_COUNT').value
 
+    @property
+    def stop_time(self):
+        return spice.sct2e(self.spacecraft_id, self._ending_ephemeris_time)
+
+    @property
+    def _starting_ephemeris_time(self):
```
Comment (jessemapel) [inline ale/drivers/kaguya_drivers.py]: Should be public, not privatel so that is properly overrides.

File: ale/base/label_pds3.py (hunk 3 above)
Comment (jessemapel) [reply ale/base/label_pds3.py]: `spacecraft_clock_start_count` seems like the logical choice to me.

**Pull request merged**